src/target/riscv/riscv.h

```diff
@@ -191,6 +191,9 @@ struct riscv_info {
 	/* The configured approach to translate virtual addresses to physical */
 	riscv_virt2phys_mode_t virt2phys_mode;
 
+	/* Halt group may be associated with an external trigger */
+	unsigned int external_trigger;
```

Comment:
As pointed out in another comment, there can be multiple external triggers in a halt (or resume) group.

The code, on the other hand, seems to force 1:1 relationship.